simulariumio/converter.py

```diff
@@ -279,6 +280,35 @@ def _determine_plot_reader(plot_type: str = "scatter") -> [Reader]:
 
         raise UnsupportedPlotTypeError(plot_type)
 
+    def _check_agent_ids_are_unique_per_frame(self) -> bool:
+        """
+        For each frame, check that none of the unique agent IDs overlap
+        """
+        bundleData = self._data["spatialData"]["bundleData"]
+        for t in range(len(bundleData)):
+            data = bundleData[t]["data"]
+            next_uid_index = 1
+            uids = []
+            get_n_subpoints = False
+            for i in range(len(data)):
+                if i == next_uid_index:
+                    # get the number of subpoints
+                    # in order to correctly increment next_uid_index
+                    if get_n_subpoints:
+                        next_uid_index += data[i] + 2
+                        get_n_subpoints = False
+                        continue
+                    # there should be a unique ID at this index, check for duplicate
+                    uid = data[i]
+                    if uid in uids:
+                        raise Exception(
+                            f"found duplicate ID {uid} in frame {t} at index {i}"
+                        )
+                    uids.append(uid)
+                    next_uid_index += 9
```

Comment:
Where does this 9 come from? Is this something like the number of numeric entries per agent?  If we ever change the data format, this will have to change. It should at least be a symbolic global constant like `OFFSET_TO_NEXT_UID_V1=9 `but even better it could be the length of a list of all the per-agent data values `len(v1_data_struct=[X,Y,Z,RX,RY,RZ, ...])`

Reply:
Also, less important for now, but since each frame is independent, maybe this function could act only on a single frame t at a time, and could run inside the read() function or in the write() function?   I suppose this would be more of an optimization for very large trajectories, to not have to loop over all the data an extra time.